src/kaleidoscope/plugin/FirmwareDump.cpp

```diff
@@ -15,6 +15,8 @@
  * this program. If not, see <http://www.gnu.org/licenses/>.
  */
 
+#ifndef ARDUINO_VIRTUAL
```

Comment:
Firmware dump does not make sense for virtual builds and is not possible to be used there anyway.